src/renderer/html_handlebars/helpers/toc.rs

```diff
@@ -40,48 +40,49 @@ impl HelperDef for RenderToc {
 
             if level > current_level {
                 while level > current_level {
-                    try!(rc.writer.write_all("<li>".as_bytes()));
-                    try!(rc.writer.write_all("<ul class=\"section\">".as_bytes()));
+                    rc.writer.write_all("<li>".as_bytes())?;
+                    rc.writer.write_all("<ul class=\"section\">".as_bytes())?;
                     current_level += 1;
                 }
-                try!(rc.writer.write_all("<li>".as_bytes()));
+                rc.writer.write_all("<li>".as_bytes())?;
             } else if level < current_level {
                 while level < current_level {
-                    try!(rc.writer.write_all("</ul>".as_bytes()));
-                    try!(rc.writer.write_all("</li>".as_bytes()));
+                    rc.writer.write_all("</ul>".as_bytes())?;
+                    rc.writer.write_all("</li>".as_bytes())?;
                     current_level -= 1;
                 }
-                try!(rc.writer.write_all("<li>".as_bytes()));
+                rc.writer.write_all("<li>".as_bytes())?;
             } else {
-                try!(rc.writer.write_all("<li".as_bytes()));
+                rc.writer.write_all("<li".as_bytes())?;
                 if item.get("section").is_none() {
-                    try!(rc.writer.write_all(" class=\"affix\"".as_bytes()));
+                    rc.writer.write_all(" class=\"affix\"".as_bytes())?;
                 }
-                try!(rc.writer.write_all(">".as_bytes()));
+                rc.writer.write_all(">".as_bytes())?;
             }
 
             // Link
             let path_exists = if let Some(path) = item.get("path") {
                 if !path.is_empty() {
-                    try!(rc.writer.write_all("<a href=\"".as_bytes()));
+                    rc.writer.write_all("<a href=\"".as_bytes())?;
 
                     // Add link
-                    try!(rc.writer.write_all(Path::new(item.get("path")
+                    rc.writer
+                        .write_all(Path::new(item.get("path")
                                                        .expect("Error: path should be Some(_)"))
                                              .with_extension("html")
                                              .to_str()
                                              .unwrap()
                                              // Hack for windows who tends to use `\` as separator instead of `/`
                                              .replace("\\", "/")
-                                             .as_bytes()));
+                                             .as_bytes())?;
```

Comment:
This bit is also a little weird, I think this can be easily fixed by moving the `Path::new()...` to a temporary variable just above.